### ISSUE TITLE
Dependency conflict on Microsoft.Extensions.Logging 1.1.1 -> 2.0.0

### ISSUE THREAD

**Issue by eloekset** [Azure/diagnostics-eventflow#182]
I've got a Service Fabric project and an Azure Function sharing the same class libraries, and I've made a dependency on EventFlow for logging. It works fine for the Service Fabric project, but the Azure Function project crashes whenever the function is triggered with the following message:

*Method not found: 'Microsoft.Extensions.Logging.ILoggerFactory Microsoft.Diagnostics.EventFlow.Inputs.LoggerFactoryExtensions.AddEventFlow(Microsoft.Extensions.Logging.ILoggerFactory, Microsoft.Diagnostics.EventFlow.DiagnosticPipeline)'.*
![image](https://user-images.githubusercontent.com/791721/36314325-bedd64aa-1334-11e8-876b-2e350ef19022.png)

I've debugged this by creating a new Azure Function project with EventFlow, which worked fine until I added a dependency on **Microsoft.Extensions.Logging 2.0.0**.

The code that raises the exception is:
```
// Configure logging
var loggerFactory = new LoggerFactory()
      .AddEventFlow(diagnosticsPipeline);
```

In my real project I get an implicit dependency on 2.0.0, since I'm using **Microsoft.EntityFrameworkCore 2.0.1**. If I try to downgrade to **Microsoft.Extensions.Logging 1.1.1** I get the following error:
![image](https://user-images.githubusercontent.com/791721/36314251-8864cddc-1334-11e8-9e7b-6488adedef76.png)

I've published the sample app where I reproduced the error here: https://github.com/eloekset/EventFlowFunctionApp1/tree/master
And this branch has a dependency on **Microsoft.Extensions.Logging 1.1.1**: https://github.com/eloekset/EventFlowFunctionApp1/tree/feature/FixRuntimeError

I haven't yet figured out why the Service Fabric project works fine, when it also has the same dependencies on the class libraries using EFCore 2.0.1. The Service Fabric project hosts a service based on ASP.NET Core having target framework set to net47, and I've set the Function App project to net47 as well.

**Comment by karolz-ms** [comment]
@eloekset can you try the workarounds specified in [.NET issue 481](https://github.com/dotnet/standard/issues/481), in particular, turning on auto binding redirect generation, and report if it helped?

**Comment by eloekset** [comment]
Thanks for the answer! I've tried to enable auto binding redirect generation. It didn't help. And Azure Function projects have host.json and local.settings.json files instead of web.config. 

I've found a blog post about explicit binding redirect for Azure Function, which helped me with a similar issue in another Azure Function project. I published a branch with this implementation to my sample repo now: https://github.com/eloekset/EventFlowFunctionApp1/tree/feature/ExplicitBindingRedirect I'll keep it for the records, but it didn't solve the issue.

I'm going to build a version of `Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging` compiled against `Microsoft.Extensions.Logging 2.0.0` and publish it on our private MyGet feed. If that doesn't fix the problem, I'm simply going to drop the Azure Function and move the functionality inside the microservice. I'll report back if building a version for 2.0.0 makes any difference.

**Comment by eloekset** [comment]
Unfortunately I'm not able to build a .NET Standard 2.0 version because of a missing `FilterParser.peg.cs` in the **Microsoft.Diagnostic.EventFlow.Core** project.

I'll give up the Azure Function project for now, and try to move that functionality inside the microservice project instead. For some reason that project works just fine, even if it's an ASP.NET Core application targeting .NET Framework 4.7, and I've got dependencies on .NET Standard 2.0 projects there as well.

**Comment by yantang-msft** [comment]
@eloekset Try build this project https://github.com/Azure/diagnostics-eventflow/tree/master/src/Microsoft.Diagnostics.EventFlow.FilterParserGenerator and the missing file should be there.

**Comment by eloekset** [comment]
I tried to build some NuGet-packages that support .NET Standard 2.0 as well as 1.6 and update my sample repo, but I didn't get the assemblies signed, so the Azure Function app fails to load them.
![image](https://user-images.githubusercontent.com/791721/36563725-57066b1a-181b-11e8-887d-47d216a113fd.png)

Anyway, I ended up dropping the Azure Function app and moved the Service Bus trigger inside a Service Fabric service as a workaround, so I'm no longer in a hurry to get this fixed.

I opened a PR with the work as far as I got. Just close it if it's not useful.

**Comment by karolz-ms** [comment]
@eloekset this is very helpful and we here at MS really appreciate your effort.

I am going to try to get to the bottom of this... just probably not this week. So I am going to keep the issue and the PR open for now. Thanks again!

**Comment by karolz-ms** [comment]
Should be fixed now with 1.4 series release on Nuget. Thanks again for help!!
